PMM-4804 Updated remove service params validation

percona-csalguero opened the pull request:
Service Name param was flagged as required but pmm-managed can retrieve a service from the DB using either service-id or service-name so, only one of them is really required.
- Removed Required() from --service-name
- Added a new validation func to reflect this behavior

percona-csalguero committed:
PMM-4804 Remove service params validation

Service ID param was flagged as required but pmm-managed can retrieve a
service from the DB using either service-id or service-name so, only one
of them is really required.
- Removed Requiered() from --service-name
- Added a new validation func to reflect this behavior

percona-csalguero committed:
Merge branch 'master' into PMM-4804-Remove-service-params

AlekSi commented on commands/management/remove.go:
```diff
@@ -68,6 +72,13 @@ func (cmd *removeMySQLCommand) serviceType() *string {
 
+func (cmd *removeMySQLCommand) validateParams() error {
+	if cmd.ServiceID == "" && cmd.ServiceName == "" {
+		return fmt.Errorf("You need to specify --service-id or --service-name")
```
pmm-managed also does not allow both: https://github.com/percona/pmm-managed/blob/91f6507da2959d972987b09a32cd710afdfaffe4/services/management/service.go#L116-L124 That's why, in general, we prefer server-side validation to client-side.

On the other side, pmm-managed does not know pmm-admin flag names, so an error message can't be perfect.

I don't know the general solution, but in that case, let's sync client-side validation implementation with pmm-managed's, and add a comment reminding us to do that on the next change.

percona-csalguero replied:
Since pmm-managed is already doing the validation I just wanted to add a pre-validation to avoid an unnecessary request but you are right, it is better to have validations just in one place.. I am going to remove this one in favor of pmm-managed's validation.

AlekSi commented on commands/management/remove_test.go:
```diff
@@ -0,0 +1,31 @@
+		cmd := removeMySQLCommand{}
+		res, err := cmd.Run()
+		assert.Nil(t, res)
+		assert.Error(t, err)
```
Let's use `EqualError`.

percona-csalguero committed:
PMM-4804 Removed client side validation

Removed client side validation in favor of pmm-managed request validation.
Removed a test to avoid needing pmm-managed running.
